File: 0001-two-sum/solution.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> map = new Dictionary<int,int>();

        for(int i = 0; i < nums.Length; i++){
            int complement = target - nums[i];
            if(map.ContainsKey(complement)){
                return new int[]{i, map[complement]};
            }
            else {
                if(!map.ContainsKey(nums[i]))map.Add(nums[i], i);
            }
        }

        return new int[]{};
    }
}

File: 0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        
        int cop;
        for(int i = 0; i < nums.length; i++){
            cop = target - nums[i];
            if(map.containsKey(cop)){
                return new int[] {map.get(cop),i};
            }
            else map.put(nums[i], i);
        }   
        return new int[]{};
    }
}
